### PR TITLE
[Core][NR-Strategy] adding iteration-number print

### PR DESCRIPTION
**Pull request by philbucher**
I think this is really handy, no clue why it was not there before

additionally removing the rank-checks since now the MPI-Logger is implemented

### CODE REVIEW

File: kratos/solving_strategies/strategies/residualbased_newton_raphson_strategy.h [hunk 1]
```diff
@@ -1039,6 +1038,10 @@ class ResidualBasedNewtonRaphsonStrategy
         TSystemVectorType& rDx = *mpDx;
         TSystemVectorType& rb  = *mpb;
 
+        if (this->GetEchoLevel() != 0){
+            KRATOS_INFO("NR-Strategy") << "Iteration " << IterationNumber << " / " << mMaxIterationNumber << std::endl;
```
Comment (loumalouomega) [inline kratos/solving_strategies/strategies/residualbased_newton_raphson_strategy.h]: I don't agree with this. It fills the simulation with a lot of noise. Many developers already complain about all the noise like @AlejandroCornejo (and myself). I think is smarter to print it during convergence criteria along with the convergence check (I mean on the same line)

**Review by loumalouomega — CHANGES_REQUESTED** [review]
Let's wait for the full discussion

File: kratos/solving_strategies/strategies/residualbased_newton_raphson_strategy.h (hunk 1 above)
Comment (philbucher) [reply kratos/solving_strategies/strategies/residualbased_newton_raphson_strategy.h]: I see your point and generally agree
Thats why you can disable it, it is protected by the echo-lvl just like the other stuff
It is just as important as the convergence information itself

How would you print from the ConvergenceCriteria, it does not know (and also should not know IMO) abt the max number of iterations=?

Comment (loumalouomega) [reply kratos/solving_strategies/strategies/residualbased_newton_raphson_strategy.h]: No, but the maximum number of iterations is not important (at least I don't consider it important). In any case a new variable stored in the ProcessInfo can be created

Comment (philbucher) [reply kratos/solving_strategies/strategies/residualbased_newton_raphson_strategy.h]: I disagree, I think it is important 
But lets see what others say

Thought this was an easy PR but I guess we have to bother the committee :)

Comment (loumalouomega) [reply kratos/solving_strategies/strategies/residualbased_newton_raphson_strategy.h]: I don't see as important. If does not converge a message will be printed. Then you know the maximum number of iterations.

Comment (philbucher) [reply kratos/solving_strategies/strategies/residualbased_newton_raphson_strategy.h]: I think it is important to know how good your problem converges, how would you plot a convergence-graph without this?

Why is the output noisy? I think with a low echo-lvl there is hardly any output...
Could you please explain what output you don't like? Maybe we can find a global solution for this then

Comment (loumalouomega) [reply kratos/solving_strategies/strategies/residualbased_newton_raphson_strategy.h]: You adding a  new line of output per each NL iteration. That's a lot of verbosity

Comment (loumalouomega) [reply kratos/solving_strategies/strategies/residualbased_newton_raphson_strategy.h]: If it is important the maximum number of NL can be sabed in a variable in the ProcessInfo

File: kratos/solving_strategies/strategies/residualbased_newton_raphson_strategy.h [hunk 2]
```diff
@@ -788,8 +786,13 @@ class ResidualBasedNewtonRaphsonStrategy
         }
 
         //plots a warning if the maximum number of iterations is exceeded
-        if (iteration_number >= mMaxIterationNumber && BaseType::GetModelPart().GetCommunicator().MyPID() == 0)
+        if (iteration_number >= mMaxIterationNumber) {
             MaxIterationsExceeded();
+        } else {
+            KRATOS_INFO_IF("NR-Strategy", this->GetEchoLevel() > 0)
+                << "Convergence achieved after " << iteration_number << " / "
+                << mMaxIterationNumber << " iterations" << std::endl;
```
Comment (philbucher) [inline kratos/solving_strategies/strategies/residualbased_newton_raphson_strategy.h:794]: what do you think now @loumalouomega 
Now it is only printed once per SolutionStep (I wanted it like this initially but I somehow messed up)

File: kratos/solving_strategies/strategies/residualbased_newton_raphson_strategy.h [hunk 3]
```diff
@@ -1064,17 +1067,13 @@ class ResidualBasedNewtonRaphsonStrategy
...
 
     virtual void MaxIterationsExceeded()
     {
-        if (this->GetEchoLevel() != 0 && BaseType::GetModelPart().GetCommunicator().MyPID() == 0)
-        {
-            std::cout << "***************************************************" << std::endl;
-            std::cout << "******* ATTENTION: max iterations exceeded ********" << std::endl;
-            std::cout << "***************************************************" << std::endl;
-        }
+        KRATOS_INFO_IF("NR-Strategy", this->GetEchoLevel() > 0)
```
Comment (sunethwarna) [inline kratos/solving_strategies/strategies/residualbased_newton_raphson_strategy.h:1074]: Wouldn't it be nicer to have this printed irrespective of the echo level as a warning?

Comment (philbucher) [reply kratos/solving_strategies/strategies/residualbased_newton_raphson_strategy.h:1074]: we can, but @loumalouomega was already saying that we have too much output

Btw the python-solvers (at least fluid & structure) do already print a warning like this (independent of echo-lvl):
~~~
::[MechanicalSolver]::  Solver did not converge for step 1
~~~

**Review by loumalouomega — APPROVED** [review]
:)

**Pull request merged**